src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4035,10 +4037,18 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers)
                 {
                     case SyntaxKind.ThisKeyword:
                         modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureExtensionMethod);
+                        if (seenRefModifier)
+                        {
+                            modifier = CheckFeatureAvailability(modifier, MessageID.IDS_FeatureRefExtensionMethods);
```

Comment:
I'm not sure why we split the 7.2 ref work into 4 features. Do we expect anyone to turn on those features one by one?
Consider consolidating.
Otherwise, maybe we should test with some features on and others off.